subsys/net/lib/app/init.c

```diff
@@ -113,7 +113,21 @@ static void setup_ipv4(struct net_if *iface)
 		return;
 	}
 
+#if defined(CONFIG_NET_DHCPV4)
+	/* In case DHCP is enabled, make the static address tentative,
+	 * to allow DHCP address to override it. This covers a usecase
+	 * of "there should be a static IP address for DHCP-less setups",
+	 * but DHCP should override it (to use it, NET_IF_MAX_IPV4_ADDR
+	 * should be set to 1). There is another usecase: "there should
+	 * always be static IP address, and optionally, DHCP address".
+	 * For that to work, NET_IF_MAX_IPV4_ADDR should be 2 (or more).
+	 * (In this case, an app will need to bind to the needed addr
+	 * explicitly.)
+	 */
+	net_if_ipv4_addr_add(iface, &addr, NET_ADDR_OVERRIDABLE, 0);
```

Comment:
Do we really need this new flag, I mean we only set it when DHCP is enabled and when we are checking things, we check if we have DHCP anyway, why MANUAL flag would not be enough here?

Reply:
> Adding new flag is cheap ;-). (And all if's, etc. are there regardless of which flag is used).

Indeed, adding flags is easy. It is just when trying to figure out the combination of these that can be problematic. Anyway, we can certainly add this new flag and remove it later if there are problems (although I am not expecting any problems here).
@erwango had a comment, so if you address that we can proceed with this patch.